server/resp/src/main/java/org/infinispan/server/resp/commands/list/LMPOP.java

```diff
@@ -108,30 +108,30 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
          }
       }
 
-      return asyncCalls(CompletableFutures.completedNull(), null, listNames.iterator(), count, isLeft, ctx, handler);
+      CompletionStage<List<Object>> cs = asyncCalls(CompletableFutures.completedNull(), null, listNames.iterator(), count, isLeft, ctx, handler);
+      return handler.stageToReturn(cs, ctx, (res, buff) -> {
+         if (res == null) Consumers.GET_ARRAY_BICONSUMER.accept(null, buff);
```

Comment:
This should really be part of the LMPOP_BICONSUMER so we can just pass `Consumers.LMPOP_BICONSUMER` to the method.

So just add the following to the LMPOP_BICONSUMER.

```java
      if (res == null) {
         GET_ARRAY_BICONSUMER.accept(null, alloc);
         return;
      }

Reply:
I'll take the chance and do the same for all the commands we're updating.